presto-main/src/main/java/com/facebook/presto/execution/scheduler/SqlQueryScheduler.java

```diff
@@ -354,11 +353,9 @@ else if (partitioningHandle.equals(SCALED_WRITER_DISTRIBUTION)) {
                 List<Node> stageNodeList;
                 if (plan.getSubStages().isEmpty()) {
                     // no remote source
-                    boolean preferDynamic = groupedExecutionForStage && isDynamicSchduleForGroupedExecution(session);
```

Comment:
There are too many `if/else`s. Let me think a bit if there is a way to merge `preferDynamic` into `groupedExecutionForStage` to avoid interface change. Flag `preferDynamic` seems to specific for `getBucketNodeMap`